pentasharp/Controllers/BusSchedule.cs

```diff
@@ -30,24 +31,44 @@ public BusScheduleController(AppDbContext context, IMapper mapper)
         [HttpPost("AddRoute")]
         public IActionResult AddRoute([FromBody] AddRouteViewModel model, int hours, int minutes)
         {
-            if (!ModelState.IsValid || string.IsNullOrWhiteSpace(model.FromLocation) || string.IsNullOrWhiteSpace(model.ToLocation))
-                return BadRequest(new { success = false, message = "Invalid data provided." });
+            var isInvalidData = !ModelState.IsValid ||
+                                string.IsNullOrWhiteSpace(model.FromLocation) ||
+                                string.IsNullOrWhiteSpace(model.ToLocation);
 
-            if (_context.BusRoutes.Any(r => r.FromLocation == model.FromLocation && r.ToLocation == model.ToLocation))
-                return BadRequest(new { success = false, message = "Route already exists." });
+            if (isInvalidData)
+            {
+                return BadRequest(new
+                {
+                    success = false,
```

Comment:
By standardized response i meant:
Create a separate class thats named something like StandardResponse what would look similar: 
public class StandardResponse<T>
{
    public bool Success { get; set; }
    public string Message { get; set; }
    public string Code { get; set; }
    public T Data { get; set; }
}

crate an Enum to define common codes and map them in the messages 
public enum ResponseCodes
{
    Success,
    InternalServerError,
    InvalidData,
    NotFound
}

public static class ResponseMessages
{
    public const string InternalServerError = "An unexpected error occurred.";
    public const string InvalidData = "Invalid data provided.";
    public const string NotFound = "Resource not found.";
}

Implement a generic factory method that uses the enums to generate responses dynamically, and then use it similarly to
if (isInvalidData)
{
    var errorResponse = BuildProviderResponse<object>(ResponseCodes.InvalidData);
    return BadRequest(errorResponse); // Wrap in BadRequest to send as HTTP 400
}
 
 fix this approach throughout the places I left that comment. 
 Note: dont just copy and paste what i said, check if you agree on it being a better approach